tests/engine/pieces/pawn.spec.js

```diff
@@ -42,6 +42,52 @@ describe('Pawn', () => {
             moves.should.be.empty;
         });
 
+        it('cannot move if there is a piece in front', () => {
+            const pawn = new Pawn(Player.WHITE);
+            const blockingPiece = new Rook(Player.BLACK);
+            board.setPiece(Square.at(4, 3), pawn);
+            board.setPiece(Square.at(5, 3), blockingPiece);
+
+            const moves = pawn.getAvailableMoves(board);
+            moves.should.be.empty;
+        });
+
+        it('cannot move two squares if there is a piece two squares in front', () => {
+            const pawn = new Pawn(Player.WHITE);
+            const blockingPiece = new Rook(Player.BLACK);
+            board.setPiece(Square.at(1, 3), pawn);
+            board.setPiece(Square.at(3, 3), blockingPiece);
+
+            const moves = pawn.getAvailableMoves(board);
+            moves.should.deep.include(Square.at(2, 3));
```

Comment:
Similarly to the last comment, does this check that there is not a move in 2 rows?